rfcs/react-components/components/simple-component-implementation.md

```diff
@@ -0,0 +1,124 @@
+# RFC: Component recipes
+
+---
+
+@sopranopillow
+
+## Summary
+
+Decide how to implement simple components that could be built with already created components.
+
+## Background
+
+Currently Persona's spec calls for a media slot that allows the use of an Avatar, image, icon, and a PresenceBadge. When implementing Persona we considered what Persona provides, it is a grid that aligns its text and media based on the given label position. This is also known as a media object by other libraries.
+
+There's one key aspect that Persona adds that makes creating the component compelling, unifying the look and feel across apps. To achieve this, we don't need the image and icon media options, we only need the Avatar and PresenceBadge (it also makes more sense that a persona only showcases person-related information). This then is the reason why the already implemented Persona only supports these slots. But there's still the question about how to add the look and feel of Fluent to media objects that have an image or icon.
+
+This is where this RFC will decide how to proceed with this portion of the problem.
+
+## Problem statement
+
+As we start working on new components, there might be some components that are simple and could be built with a simple layout and the already finished components. When creating these components, we currently don't have a guidance on whether to let the user build the component or us provide the component. This RFC offers three options, one that will require the user to build the components using instructions (recipes), another where we will provide the implementation of the component, and a combination of both option 1 and 2 providing a partial implementation and a recipe.
+
+#### What makes a component a candidate for this RFC's solution?
+
+- Its implementation is simple.
+- The desired design is achievable adjusting the props of the inner components.
+- It doesn't have complex accessibility requirements.
+- It doesn't have strict design requirements.
+
```

Comment:
Consider detailing the core considerations you think the team should use to make a decision of one approach vs. another.  Here's some examples:

* The decision should encourage components that provide high value beyond general purpose layout.  

For example, v9 deprecated and did not reimplement Stack.  Stack provides low value over using flexbox directly and was very general purpose.

* The decision should encourage components that are general purpose enough to be core to Fluent

Components like Field provide label, error message, and input component layout and coordination that can be used in many form scenarios. Components like a SharePoint file-list specific Card might not be reusable by others easily and maybe should live outside the core of Fluent and be implemented by the team that uses them.  This brings up a secondary question of should our decision favor a pattern that other developers building on top of fluent should use?

* The decision should prefer a solution that builds on top of our composition model.

We have a composition models that leverages slots, hooks, and context to make customization and replacement possible.  Which solutions leverage the existing composition model the best to maintain developer productivity and extensibility.